src/Chapter4.hs

```diff
@@ -489,6 +495,18 @@ Implement the 'Applicative' instance for our 'List' type.
   type.
 -}
 
+instance Applicative List where
+    pure :: a -> List a
+    pure x = Cons x Empty
+
+    (<*>) :: List (a -> b) -> List a -> List b
+    Empty <*> _ = Empty
+    _ <*> Empty = Empty
+    Cons f fs <*> l = append (fmap f l) (fs <*> l)
+
+append :: List a -> List a -> List a
+append Empty ys = ys
```

Comment:
Perfect!

You can one more case to make your solution even more efficient:

```haskell
append xs Empty = xs
```

In that case, you won't need to walk through each element of the first list when the second one is empty and return the list instantly 🚤